pr-dhl-woocommerce/includes/REST_API/Paket/Client.php

```diff
@@ -253,31 +254,39 @@ protected function request_pickup_info_to_request_data( Pickup_Request_Info $req
 
 		//Pickup location & business hours
 		$pickup_location_array = array(
+			"type" => "Address",
 			'pickupAddress' 	=> array(
 				'name1' 		=> $request_pickup_info->pickup_contact['name'],
-				'nativeAddress' 	=> $request_pickup_info->pickup_address
+				'name2'			=> '',
+				'addressStreet' => $request_pickup_info->pickup_address['addressStreet'],
+				'addressHouse'  => $request_pickup_info->pickup_address['addressHouse'],
+				'city'			=> $request_pickup_info->pickup_address['city'],
+				'postalCode' 	=> $request_pickup_info->pickup_address['postalCode'],
+				'state' 		=> $request_pickup_info->pickup_address['state'],
+				'country'		=> $request_pickup_info->pickup_address['country']
 			)
 		);
-		$business_hours = null;
-		if ( $request_pickup_info->business_hours ) {
-			$pickup_location_array['businessHours']	= $request_pickup_info->business_hours;
-		}
 
 		$request_data = array(
 			'pickupLocation' 	=> $pickup_location_array,
 			'pickupDetails'	  => array(
 				'pickupDate' 	=> $pickup_info_array,
-				'emailNotification'	=> '',
 				'totalWeight'		=> $request_pickup_info->pickup_details['weight'],
 				'comment'			=> '',
 			),
 			'shipmentDetails'	  => array(
 				'shipments' 	=> $request_pickup_info->shipments
 			),
 			'contactPerson'	  => array(
-				'name' 	=> $request_pickup_info->pickup_contact['name'],
-				'phone' 	=> $request_pickup_info->pickup_contact['phone'],
-				'email' 	=> $request_pickup_info->pickup_contact['email'],
+				array(
+					'name' 			   => $request_pickup_info->pickup_contact['name'],
+					'phone' 		   => $request_pickup_info->pickup_contact['phone'],
+					'email' 		   => $request_pickup_info->pickup_contact['email'],
+					"emailNotification"=> array(
+						"sendPickupConfirmationEmail" => "true",
+						"sendPickupTimeWindowEmail"	  => "true"
```

Comment:
And I'm also wondering why you set this to "true"? can you check this with Shadi, is there any setting for this? is it required? I can see it was disabled in the previous code.

Reply:
When tested it with the one used before , it gives error 400 so these values are mandatory